platform/migration/platform-migratable/src/main/java/org/codice/ddf/migration/impl/PlatformMigratable.java

```diff
@@ -0,0 +1,216 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration.impl;
+
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.migration.Migratable;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationMetadata;
+import org.codice.ddf.migration.MigrationWarning;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class handles the export process for all Platform configurations and system files.
+ *
+ */
+public class PlatformMigratable implements Migratable {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(PlatformMigratable.class);
+
+    private static final String KEYSTORE_SYSTEM_PROP = "javax.net.ssl.keyStore";
+
+    private static final String TRUSTSTORE_SYSTEM_PROP = "javax.net.ssl.trustStore";
+
+    private static final String WS_SECURITY_DIR = "etc/ws-security";
+
+    private static final String SYSTEM_PROPERTIES = "etc/system.properties";
+
+    private static final String USERS_PROPERTIES = "etc/users.properties";
+    
+    private static final String ABSOLUTE_PATH_WARNING =
+            "The value for property [%s] is set to a path [%s] that is absolute; "
+                    + "therefore, the file will not be included in the export.  "
+                    + "Check that the file exists on the system you're migrating to "
+                    + "or update the property value and export again.";
+
+    private static final String OUTSIDE_PATH_WARNING =
+            "The value for property [%s] is set to a path [%s] that is outside [%s]; "
+                    + "therefore, the file will not be included in the export.  "
+                    + "Check that the file exists on the system you're migrating to "
+                    + "or update the property value and export again.";
+
+    private static final String UNREAL_PATH_WARNING =
+            "The value for property [%s] is set to a path [%s] that could not coerced into a real path; "
+                    + "therefore, the file will not be included in the export.  "
+                    + "Check that the file exists on the system you're migrating to "
+                    + "or update the property value and export again.";
+
+    private final Path ddfHome;
+    
+    public PlatformMigratable(@NotNull Path ddfHome) {
+        notNull(ddfHome, "ddfHome cannot be null");
+        this.ddfHome = ddfHome;
+    }
+    
+    @Override
```

Comment:
Ah! I was having a hard time understanding the comment :-)

Interface comments are inherited by default when you don't put any comments on the class that implements it. We can repeat the annotations on the implementing classes if you think it's better. I'm ok with that, as far as we make sure we keep them in-sync :-)